File: mobile/lib/models/zone.dart
class Zone {
  final int id;
  final String nom;
  final double superficie;
  final String? culture;
  final double? seuilHumidite;
  final int? espaceId;
  final String? espaceNom;
  final bool active;
  final double? derniereHumidite;
  final String? derniereMesure;

  Zone({
    required this.id,
    required this.nom,
    required this.superficie,
    this.culture,
    this.seuilHumidite,
    this.espaceId,
    this.espaceNom,
    required this.active,
    this.derniereHumidite,
    this.derniereMesure,
  });

  factory Zone.fromJson(Map<String, dynamic> json) {
    return Zone(
      id: json['id'],
      nom: json['nom'],
      superficie: (json['superficie'] as num).toDouble(),
      culture: json['culture'],
      seuilHumidite: (json['seuilHumidite'] as num?)?.toDouble(),
      espaceId: json['espaceId'],
      espaceNom: json['espaceNom'],
      active: json['active'] ?? true,
      derniereHumidite: (json['derniereHumidite'] as num?)?.toDouble(),
      derniereMesure: json['derniereMesure'],
    );
  }
}

File: backend/src/main/java/com/agricol/model/Zone.java
package com.agricol.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "zones")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Zone {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nom;

    @Column(nullable = false)
    private Double superficie;

    private String culture;

    @Column(name = "seuil_humidite")
    private Double seuilHumidite;

    @Column(name = "vanne_gpio")
    private Integer vanneGpio;

    @Column(name = "capteur_gpio")
    private Integer capteurGpio;

    @Column(name = "espace_id")
    private Long espaceId;

    @Column(name = "active")
    private Boolean active;
}
